Documentation/CodingGuidelines/CodingBestPractices/NamedArguments.rst

```diff
@@ -0,0 +1,247 @@
+    :caption: Value object using public constructor property promotion
+
+    final readonly class Label implements \JsonSerializable
+    {
+        public function __construct(
+            public string $label,
+            public string $color = '#ff8700',
+            public int $priority = 0,
+        ) {}
+
+        public function jsonSerialize(): array
+        {
+            return get_object_vars($this);
+        }
+    }
+
+Using public constructor property promotions (PCPP) facilitates object
+initialization, representing one of the primary use cases for named arguments
+envisioned by PHP developers:
+
+.. code-block:: php
+    :caption: Instantiate a PCPP value object using named arguments
+
+    $label = new Label(
+        label: $myLabel,
+        color: $myColor,
+        priority: -1,
+    );
+
+Objects with such class signatures MUST be instantiated using named arguments to
+maintain API consistency. Standardizing named argument usage allows the TYPO3
+core to introduce deprecations for argument removals seamlessly.
+
+Invoking 2nd Party Dependency Methods
+-------------------------------------
+
+The TYPO3 core refrains from employing named arguments when calling library
+code from dependent packages unless the library explicitly mandates such usage
+and defines its variable names as part of the API, a practice seldom observed
+currently.
+
+As package consumer, the TYPO3 core must assume that packages don’t treat their
+variable names as API, they may change anytime. If TYPO3 core would use named
+arguments for library calls, this may trigger regressions: Suppose a patch level
+release of a library changes a variable name of some method that we call using
+named arguments. This would immediately break when TYPO3 projects upgrade to
+this patch level release due to the power of semantic versioning. TYPO3 core must
+avoid this scenario.
+
+Invoking core API
+-----------------
+
+Within the TYPO3 core, named arguments are not used when invoking its own methods.
+There are exceptions in specific scenarios as outlined below, however these are
+the reasons for not using named arguments:
+
+* TYPO3 core tries to be as consistent as possible
+
+* Setting a good example for extensions
+
+* Avoiding complications and side effects during refactoring
+
+* Addressing legacy code within the TYPO3 core containing methods with less-desirable
+  variable names, aiming for gradual improvement without disruptions
+
+* Preventing issues with inheritance, especially in situations like this:
+
+  .. code-block:: php
+      :caption: PHP error using named arguments and inheritance
+
+      interface I {
+          public function test($foo, $bar);
+      }
+
+      class C implements I {
+          public function test($a, $b) {}
+      }
+
+      $obj = new C();
+
+      // Pass params according to I::test() contract
+      $obj->test(foo: "foo", bar: "bar"); // ERROR!
+
+Utilizing Named Arguments in PHPUnit Test Data Providers
```

Comment:
```suggestion
Utilizing named arguments in PHPUnit test Data Providers
```

(not sure, but I'd capitalize "Data Provider" as a term)

Reply:
Hmm, would not capitalize it: PHPUnit itself uses it lowercased in a sentence: https://docs.phpunit.de/en/11.0/writing-tests-for-phpunit.html#data-providers

Reply:
i don't care. feel free.